src/components/StorageOverview/DataResiliency/DataResiliency.js

```diff
@@ -0,0 +1,100 @@
+import PropTypes from 'prop-types';
+import { Icon, UtilizationBar } from 'patternfly-react';
+
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '../../Dashboard/DashboardCard';
+import { StorageOverviewContext } from '../StorageOverviewContext';
+import { InlineLoading } from '../../Loading';
+import { getCapacityStats } from '../../../selectors/prometheus/selectors';
+
+const DataResiliencyStatusBody = ({ status }) =>
+  status ? (
+    <Fragment>
+      <div className="kubevirt-data-resiliency__status-title-ok">Your Data Is Resilient</div>
+      <div className="kubevirt-data-resiliency__icon-ok">
+        <Icon type="fa" name="check-circle" size="2x" />
+      </div>
+    </Fragment>
+  ) : (
+    <Fragment>
+      <div className="kubevirt-data-resiliency__icon-error">
+        <Icon type="fa" name="exclamation-triangle" size="2x" />
+      </div>
+      <div className="kubevirt-data-resiliency__status-title-error">Some Error Occurred</div>
+    </Fragment>
+  );
+
+const DataResiliencyBuildBody = ({ progressPercentage }) => (
+  <Fragment>
+    <div className="kubevirt-data-resiliency__title">Rebuilding data resiliency</div>
+    <UtilizationBar
+      className="kubevirt-data-resiliency__utilization-bar"
+      now={progressPercentage}
+      description="Rebuilding in Progress"
+      descriptionPlacementTop
+      label={`${progressPercentage}%`}
+    />
+  </Fragment>
+);
+
+export const DataResiliency = ({ activePgRaw, totalPgRaw, LoadingComponent }) => {
+  const activePg = getCapacityStats(activePgRaw);
+  const totalPg = getCapacityStats(totalPgRaw);
+  const progressPercentage = ((activePg / totalPg) * 100).toFixed(1);
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Data Resiliency</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody
+        className="kubevirt-data-resiliency__dashboard-body"
+        isLoading={!activePg && totalPg}
+        LoadingComponent={LoadingComponent}
+      >
+        {progressPercentage % 100 === 0 || !(activePg && totalPg) ? (
+          <DataResiliencyStatusBody status={activePg && totalPg} />
+        ) : (
+          <DataResiliencyBuildBody progressPercentage={progressPercentage} />
+        )}
+      </DashboardCardBody>
+    </DashboardCard>
+  );
+};
+
+DataResiliency.defaultProps = {
+  activePgRaw: null,
+  totalPgRaw: null,
+  LoadingComponent: InlineLoading,
+};
+
+DataResiliencyStatusBody.defaultProps = {
+  status: null,
+};
+
+DataResiliencyBuildBody.defaultProps = {
+  progressPercentage: null,
+};
+
+DataResiliency.propTypes = {
+  activePgRaw: PropTypes.object,
+  totalPgRaw: PropTypes.object,
+  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+};
+
+DataResiliencyBuildBody.propTypes = {
+  progressPercentage: PropTypes.number,
+};
+
+DataResiliencyStatusBody.propTypes = {
+  status: PropTypes.bool,
```

Comment:
please use something like `isStatusSuccessful` or `isSuccess` if we do not pass a status object.

Reply:
oh right! I am bad with names..thanx!!